src/libarena/lib.rs

```diff
@@ -386,7 +386,7 @@ impl DroplessArena {
             self.align(align);
 
             let future_end = intrinsics::arith_offset(self.ptr.get(), bytes as isize);
-            if (future_end as *mut u8) >= self.end.get() {
+            if (future_end as *mut u8) > self.end.get() {
                 self.grow(bytes);
```

Comment:
`arith_offset` is the same as `<*mut T>::wrapping_offset`; it seems like this method is not really doing the correct thing if `self.ptr.wrapping_add(bytes)` wraps around? Then we don't grow but bump the pointer anyway.

Seems like alloc_raw needs to be an unsafe method with this tricky requirement. I can report a bug for this - initial impression is that `alloc_slice` is not correct for all inputs because of this bug.

Reply:
Sounds like a good follow-up!